src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -87,13 +87,35 @@ public static void ToCharsBuffer(byte value, Span<char> buffer, int startingInde
         }
 
 #if SYSTEM_PRIVATE_CORELIB
+        // Converts Vector128<byte> into 2xVector128<byte> ASCII Hex representation
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static (Vector128<byte>, Vector128<byte>) AsciiToHexVector128(Vector128<byte> src, Vector128<byte> hexMap)
+        {
+            Debug.Assert(Ssse3.IsSupported || AdvSimd.Arm64.IsSupported);
+            // The algorithm is simple: a single srcVec (contains the whole 16b Guid) is converted
+            // into nibbles and then, via hexMap, converted into a HEX representation via
+            // Shuffle(nibbles, srcVec). ASCII is then expanded to UTF-16.
+            Vector128<byte> nibbles = Vector128.ShiftRightLogical(src.AsUInt64(), 4).AsByte();
+            Vector128<byte> lowNibbles = Vector128.UnpackLow(nibbles, src);
+            Vector128<byte> highNibbles = Vector128.UnpackHigh(nibbles, src);
+
+            return (ShuffleUnsafe(hexMap, lowNibbles & Vector128.Create((byte)0xF)),
+                ShuffleUnsafe(hexMap, highNibbles & Vector128.Create((byte)0xF)));
+
+            // TODO: remove once https://github.com/dotnet/runtime/pull/80963 is merged
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            static Vector128<byte> ShuffleUnsafe(Vector128<byte> value, Vector128<byte> mask)
+                => Ssse3.IsSupported ? Ssse3.Shuffle(value, mask) : AdvSimd.Arm64.VectorTableLookup(value, mask);
```

Comment:
@radekdoulik could you take a look at this and the wasm implmentation

Reply:
NOTE: we already have this hand-made crossplat helper in 4 different places (and @MihaZupan unified them, it's that the PR is not yet merged)